pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -358,6 +358,9 @@ def __log_result_step_metrics(self, output, batch_idx):
         step_log_metrics = output.get_batch_log_metrics(include_forked_originals=False)
         step_pbar_metrics = output.get_batch_pbar_metrics(include_forked_originals=False)
 
+        cached_batch_log_metrics = self.trainer.logger_connector.cached_results(self.testing)\
```

Comment:
Just realised you use a variable to reference the testing cache, we should do this for train :)

Reply:
Yes, in `evaluation` mode, we use self.testing to differentiate between validation and test